src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -1371,13 +1377,44 @@ public void Synchronized()
         /// </summary>
         [Theory]
         [MemberData(nameof(RegexHelpers.AvailableEngines_MemberData), MemberType = typeof(RegexHelpers))]
-        public async Task Mach_Boundary_DFA(RegexEngine engine)
+        public async Task Match_Boundary(RegexEngine engine)
         {
             Regex r = await RegexHelpers.GetRegexAsync(engine, @"\b\w+\b");
             Assert.False(r.IsMatch(" AB\u200cCD "));
             Assert.False(r.IsMatch(" AB\u200dCD "));
         }
 
+        public static IEnumerable<object[]> Match_Count_TestData()
+        {
+            foreach (RegexEngine engine in RegexHelpers.AvailableEngines)
+            {
+                yield return new object[] { engine, @"\b\w+\b", "one two three", 3 };
+                yield return new object[] { engine, @"\b\w+\b", "on\u200ce two three", 2 };
+                yield return new object[] { engine, @"\b\w+\b", "one tw\u200do three", 2 };
+            }
+
+            string b1 = @"((?<=\w)(?!\w)|(?<!\w)(?=\w))";
+            string b2 = @"((?<=\w)(?=\W)|(?<=\W)(?=\w))";
+            // Lookarounds are currently not supported in the NonBacktracking engine
+            foreach (RegexEngine engine in new RegexEngine[] { RegexEngine.Compiled, RegexEngine.Interpreter })
```

Comment:
This will leave out the source generator.  Can you change it to:
```C#
foreach (RegexEngine engine in RegexHelpers.AvailableEngines)
{
    if (engine == RegexEngine.NonBacktracking) continue;
    ...
}
```
or something like that, so that we're only opting-out the engine we know doesn't work?

Reply:
sure, will update.